src/compiler/program.ts

```diff
@@ -1520,7 +1527,297 @@ namespace ts {
+                    activeLint = initializedLints[i];
+                    if (!needsReset[i]) {
+                        activeLint.accepted = true;
+                        needsReset[i] = false;
+                    }
+                    if (!activeLint.errored && activeLint.walker.afterVisit) {
+                        try {
+                            activeLint.walker.afterVisit(node, error);
+                        }
+                        catch (e) {
+                            activeLint.errored = true;
+                            diagnostics.push(createExtensionDiagnostic(errorQualifiedName("!!!"), `afterVisit failed with error: ${e}`, /*sourceFile*/undefined, /*start*/undefined, /*length*/undefined, DiagnosticCategory.Error));
+                        }
+                    }
+                }
+            }
+
+            function errorQualifiedName(shortname?: string) {
+                if (!shortname) return activeLint.name;
+                return `${activeLint.name}(${shortname})`;
+            }
+
+            function error(err: string): void;
+            function error(err: string, node: Node): void;
+            function error(err: string, start: number, length: number): void;
+            function error(shortname: string, err: string): void;
+            function error(shortname: string, err: string, span: Node): void;
+            function error(shortname: string, err: string, start: number, length: number): void;
+            function error(level: DiagnosticCategory, err: string): void;
+            function error(level: DiagnosticCategory, err: string, node: Node): void;
+            function error(level: DiagnosticCategory, err: string, start: number, length: number): void;
+            function error(level: DiagnosticCategory, shortname: string, err: string): void;
+            function error(level: DiagnosticCategory, shortname: string, err: string, span: Node): void;
+            function error(level: DiagnosticCategory, shortname: string, err: string, start: number, length: number): void;
+            function error(
+                levelErrOrShortname: DiagnosticCategory | string,
+                errShortnameSpanOrStart?: string | Node | number,
+                errSpanStartOrLength?: string | Node | number,
+                startSpanOrLength?: number | Node,
+                length?: number): void {
+                if (typeof levelErrOrShortname === "undefined") {
+                    Debug.fail("You must at least provide an error message to error.");
+                    return;
+                }
+                if (typeof errShortnameSpanOrStart === "undefined") {
+                    if (typeof levelErrOrShortname !== "string") {
+                        Debug.fail("If only one argument is passed to error, it must be the error message string.");
+                        return;
+                    }
+                    // error(err: string): void;

```

Comment:
Yes - I was placing comments with the corresponding overload next to their implementation, just for clarity.